### ISSUE TITLE
min-shards-available violations don't always result in http500

### ISSUE THREAD

**Issue by Dieterbe** [grafana/metrictank#1025]
easy to reproduce. 
apply this patch:
```
diff --git a/docker/docker-cluster/metrictank.ini b/docker/docker-cluster/metrictank.ini
index 47246f01..3282af10 100644
--- a/docker/docker-cluster/metrictank.ini
+++ b/docker/docker-cluster/metrictank.ini
@@ -233,7 +233,7 @@ peers =
 # Operating mode of cluster. (single|multi)
 mode = single
 # minimum number of shards that must be available for a query to be handled.
-min-available-shards = 0
+min-available-shards = 10
 # How long to wait before aborting http requests to cluster peers and returning a http 503 service unavailable
 http-timeout = 60s

```
then run: `./launch.sh docker-cluster`. store some fake data and try to load a chart.
3 times i loaded a chart, carefully making sure not to hit reload until the previous response had come in (and grafana showed the error)

```
metrictank0_1  | [Macaron] 2018-08-29 16:30:23: Started POST /render/ for 172.18.0.2
metrictank0_1  | 2018/08/29 16:30:23 [cluster.go:317 peerQuerySpeculative()] [E] HTTP peerQuery unable to get peers, Insufficient shards available.
metrictank0_1  | [Macaron] 2018-08-29 16:30:23: Completed /render/ 503 Service Unavailable in 448.199µs

metrictank0_1  | [Macaron] 2018-08-29 16:30:23: Started POST /render/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:30:23: Completed /render/ 499  in 332.867µs
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:30:23 +0000] "POST /render HTTP/1.1" 500 1578
grafana_1      | t=2018-08-29T16:30:23+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/datasources/proxy/1/render status=500 remote_addr=172.18.0.1 time_ms=48 size=1258 referer="http://localhost:3000/d/y9QAen2mz/metrictank?panelId=9&fullscreen&edit&orgId=1"

metrictank0_1  | [Macaron] 2018-08-29 16:30:26: Started POST /render/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:30:26: Completed /render/ 503 Service Unavailable in 437.938µs
metrictank0_1  | 2018/08/29 16:30:26 [cluster.go:317 peerQuerySpeculative()] [E] HTTP peerQuery unable to get peers, Insufficient shards available.
metrictank0_1  | [Macaron] 2018-08-29 16:30:26: Started POST /render/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:30:26: Completed /render/ 499  in 332.158µs
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:30:26 +0000] "POST /render HTTP/1.1" 500 1578
grafana_1      | t=2018-08-29T16:30:26+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/datasources/proxy/1/render status=500 remote_addr=172.18.0.1 time_ms=48 size=1258 referer="http://localhost:3000/d/y9QAen2mz/metrictank?panelId=9&fullscreen&edit&orgId=1"

```

**Comment by Dieterbe** [comment]
ditto for `find` . 3 failed finds then one succeeds:
```
metrictank0_1  | [Macaron] 2018-08-29 16:37:20: Started POST /metrics/find/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:37:20: Completed /metrics/find/ 503 Service Unavailable in 224.831µs
metrictank0_1  | 2018/08/29 16:37:20 [cluster.go:317 peerQuerySpeculative()] [E] HTTP peerQuery unable to get peers, Insufficient shards available.
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:37:20 +0000] "GET /metrics/find?from=1535474005&query=*&until=1535560765 HTTP/1.1" 500 1139
grafana_1      | t=2018-08-29T16:37:20+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/api/datasources/proxy/1/metrics/find status=500 remote_addr=172.18.0.1 time_ms=27 size=820 referer="http://localhost:3000/d/y9QAen2mz/metrictank?panelId=9&fullscreen&edit&orgId=1"


metrictank0_1  | [Macaron] 2018-08-29 16:37:26: Started POST /metrics/find/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:37:26: Completed /metrics/find/ 503 Service Unavailable in 246.531µs
metrictank0_1  | 2018/08/29 16:37:26 [cluster.go:317 peerQuerySpeculative()] [E] HTTP peerQuery unable to get peers, Insufficient shards available.
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:37:26 +0000] "GET /metrics/find?from=1535473945&query=*&until=1535560825 HTTP/1.1" 500 1139
grafana_1      | t=2018-08-29T16:37:26+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/api/datasources/proxy/1/metrics/find status=500 remote_addr=172.18.0.1 time_ms=24 size=820 referer="http://localhost:3000/d/y9QAen2mz/metrictank?panelId=9&fullscreen&edit&orgId=1"
metrictank3_1  | 2018/08/29 16:37:26 [DEBUG] memberlist: Stream connection from=172.18.0.10:43270
metrictank0_1  | 2018/08/29 16:37:26 [DEBUG] memberlist: Initiating push/pull sync with: 172.18.0.11:7946
metrictank0_1  | 2018/08/29 16:37:27 [DEBUG] memberlist: Stream connection from=172.18.0.12:53434
metrictank2_1  | 2018/08/29 16:37:27 [DEBUG] memberlist: Initiating push/pull sync with: 172.18.0.10:7946


metrictank0_1  | [Macaron] 2018-08-29 16:37:31: Started POST /metrics/find/ for 172.18.0.2
metrictank0_1  | [Macaron] 2018-08-29 16:37:31: Completed /metrics/find/ 503 Service Unavailable in 356.07µs
metrictank0_1  | 2018/08/29 16:37:31 [cluster.go:317 peerQuerySpeculative()] [E] HTTP peerQuery unable to get peers, Insufficient shards available.
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:37:31 +0000] "GET /metrics/find?from=1535473885&query=*&until=1535560885 HTTP/1.1" 500 1138
grafana_1      | t=2018-08-29T16:37:31+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=GET path=/api/datasources/proxy/1/metrics/find status=500 remote_addr=172.18.0.1 time_ms=45 size=819 referer="http://localhost:3000/d/y9QAen2mz/metrictank?panelId=9&fullscreen&edit&orgId=1"
metrictank1_1  | 2018/08/29 16:37:33 [DEBUG] memberlist: Initiating push/pull sync with: 172.18.0.11:7946
metrictank3_1  | 2018/08/29 16:37:33 [DEBUG] memberlist: Stream connection from=172.18.0.13:47620
graphite_1     | 172.18.0.6 - - [29/Aug/2018:16:37:34 +0000] "GET /metrics/find?from=1535473825&query=*&until=1535560945 HTTP/1.1" 200 289
```

**Comment by Dieterbe** [comment]
seems to be a graphite problem. sometimes graphite decides not to query MT and then it returns 200's. some kind of poorly behaving cache?

**Comment by Dieterbe** [comment]
fixed in https://github.com/raintank/graphite-mt/pull/11, published along with graphite update as 1.1.4 on https://hub.docker.com/r/raintank/graphite-mt/tags/
